opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/MonitoredServiceDaoHibernate.java

```diff
@@ -151,4 +152,16 @@ public OnmsMonitoredService get(Integer nodeId, InetAddress ipAddr, Integer ifIn
                     "where node.id = ?",
                     nodeId);
     }
+
+    @Override
+    public List<OnmsMonitoredService> findByServiceTypeAndIpInterfaceId(String serviceTypePrefix, List<Integer> ipInterfaceIds) {
+        var idParam = ipInterfaceIds.stream().map(String::valueOf).collect(Collectors.joining(","));
+
+        String query =
+            "from OnmsMonitoredService svc " +
+                "where svc.serviceType.name like '" + serviceTypePrefix + "%' " +
+                "and svc.ipInterface.id in (" + idParam + ")";
```

Comment:
Use query parameters, to avoid SQL injection.

Reply:
Query parameters for 'in' clause did not seem to work as expected, however, I can take another look. I agree we should be using query parameters.

Reply:
An alternative would be to use Hibernate's criteria queries.

Reply:
Rewritten to use CriteriaBuilder, see latest version.